### PR TITLE
Converts all .proto files to proto3 syntax.

### PR DESCRIPTION
**Pull request by cschuet**
[RFC=0000](https://github.com/googlecartographer/rfcs/blob/master/text/0000-proto3-transition.md)

### CODE REVIEW

File: cartographer/mapping/sparse_pose_graph.cc [hunk 1]
```diff
@@ -42,8 +43,9 @@ SparsePoseGraph::Constraint::Tag FromProto(
       return SparsePoseGraph::Constraint::Tag::INTRA_SUBMAP;
     case proto::SparsePoseGraph::Constraint::INTER_SUBMAP:
       return SparsePoseGraph::Constraint::Tag::INTER_SUBMAP;
+    default:
+      LOG(FATAL) << "Unsupported tag.";
```
Comment (cschuet) [inline cartographer/mapping/sparse_pose_graph.cc]: This fixes

> error: enumeration value 'SparsePoseGraph_Constraint_Tag_SparsePoseGraph_Constraint_Tag_INT_MAX_SENTINEL_DO_NOT_USE_' not handled in switch [-Werror=switch]

**Comment by SirVer** [comment]
Please make a link in the body (i.e. first comment) of this PR to make the commit message more useful.

**Comment by SirVer** [comment]
Lgtm. Please open a bug each for

1. Updating the documentation tool.
2. removing `[packed]`annotations

so we do not forget.

**Comment by SirVer** [comment]
@wally-the-cartographer merge

**Comment by wally-the-cartographer** [comment]
Merge requested by authorized user SirVer. Merge queue now has a length of 1.

**Comment by wally-the-cartographer** [comment]
Error: exit status 1

Log:
~~~
=> Rebasing PR 650 on googlecartographer/cartographer onto master.
=> Switching to branch master.
 $ git rebase --abort [data/cartographer]
No rebase in progress?
 $ git checkout -f [data/cartographer]
Your branch is up to date with 'origin/master'.
 $ git clean -fd [data/cartographer]
 $ git checkout master [data/cartographer]
Already on 'master'
Your branch is up to date with 'origin/master'.
 $ git pull [data/cartographer]
From https://github.com/googlecartographer/cartographer
   5363285..eb4415d  master     -> origin/master
Updating 5363285..eb4415d
Fast-forward
 cartographer/mapping/global_trajectory_builder.h   | 17 ++++++++----
 .../mapping_2d/local_trajectory_builder.cc         | 12 ++++++---
 cartographer/mapping_2d/local_trajectory_builder.h | 15 ++++++++---
 .../mapping_3d/local_trajectory_builder.cc         | 31 ++++++++++++----------
 cartographer/mapping_3d/local_trajectory_builder.h | 13 +++++++--
 .../mapping_3d/local_trajectory_builder_test.cc    | 13 ++++-----
 6 files changed, 67 insertions(+), 34 deletions(-)
 $ git remote add cschuet git@github.com:cschuet/cartographer.git [data/cartographer]
 $ git fetch cschuet [data/cartographer]
From github.com:cschuet/cartographer
 * [new branch]      filtering_point_processor2 -> cschuet/filtering_point_processor2
 * [new branch]      final_initial_pose -> cschuet/final_initial_pose
 * [new branch]      find               -> cschuet/find
 * [new branch]      fix                -> cschuet/fix
 * [new branch]      histograms         -> cschuet/histograms
 * [new branch]      initial_pose       -> cschuet/initial_pose
 * [new branch]      lower_bound_maop   -> cschuet/lower_bound_maop
 * [new branch]      mapbyidspg         -> cschuet/mapbyidspg
 * [new branch]      master             -> cschuet/master
 * [new branch]      opti               -> cschuet/opti
 * [new branch]      opti3d             -> cschuet/opti3d
 * [new branch]      pose_renames       -> cschuet/pose_renames
 * [new branch]      proto3_packed      -> cschuet/proto3_packed
 * [new branch]      range              -> cschuet/range
 * [new branch]      rename             -> cschuet/rename
 * [new branch]      slides             -> cschuet/slides
 * [new branch]      timeout            -> cschuet/timeout
 * [new branch]      trajectory_nodes   -> cschuet/trajectory_nodes
 * [new branch]      trimmed            -> cschuet/trimmed
 $ git branch --track pr_650 cschuet/proto3 [data/cartographer]
error: the requested upstream branch 'cschuet/proto3' does not exist
hint: 
hint: If you are planning on basing your work on an upstream
hint: branch that already exists at the remote, you may need to
hint: run "git fetch" to retrieve it.
hint: 
hint: If you are planning to push out a new local branch that
hint: will track its remote counterpart, you may want to use
hint: "git push -u" to set the upstream config as you push.
=> Abandoning branch pr_650.
=> Switching to branch master.
 $ git rebase --abort [data/cartographer]
No rebase in progress?
 $ git checkout -f [data/cartographer]
Your branch is up to date with 'origin/master'.
 $ git clean -fd [data/cartographer]
 $ git checkout master [data/cartographer]
Already on 'master'
Your branch is up to date with 'origin/master'.
 $ git branch -D pr_650 [data/cartographer]
error: branch 'pr_650' not found.

~~~

**Pull request closed without merging**